src/vi/nhattruyen/src/eu/kanade/tachiyomi/extension/vi/nhattruyen/NhatTruyen.kt

```diff
@@ -177,4 +187,32 @@ class NhatTruyen : WPComics(
             OrderFilter(),
         )
     }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val url = client.newCall(GET(baseUrl + manga.url, headers)).execute().asJsoup()
+        if (checkChapterLists(url).isNotEmpty()) {
+            val slug = manga.url.removePrefix("/truyen-tranh/")
+            return client.newCall(GET("$baseUrl/Comic/Services/ComicService.asmx/ChapterList?slug=$slug", headers))
+                .asObservableSuccess()
+                .map { response -> chapterListParse(response) }
+        }
+        return super.fetchChapterList(manga)
+    }
```

Comment:
just fetch from the ajax call all the time, instead of checking for view-more. and you return super method but you changed chapterlistparse for json parsing, so will it work?

Reply:
> just fetch from the ajax call all the time, instead of checking for view-more. and you return super method but you changed chapterlistparse for json parsing, so will it work?

It works. it’s better for the web to make the call only when there’s manga with many chapters.

Reply:
> you have changed `chapterListParse` so now there is no logic for html parsing, which you are relying on in `super.fetchChapterList`

I changed it to chapterlistRequest. Now it's complete, ready to merge 👍